src/game/Spells/Spell.cpp

```diff
@@ -6972,7 +6972,7 @@ SpellCastResult Spell::CheckCast(bool strict)
                 if (m_casterUnit->IsInWater() && (!m_casterUnit->IsPlayer() || static_cast<Player*>(m_casterUnit)->IsInHighLiquid()))
                     return SPELL_FAILED_ONLY_ABOVEWATER;
 
-                if (m_casterUnit->IsPlayer() && static_cast<Player*>(m_casterUnit)->GetTransport())
+                if (m_casterUnit->IsPlayer() && (static_cast<Player*>(m_casterUnit)->GetTransport() && static_cast<Player*>(m_casterUnit)->GetTransport()->IsMoTransport()))
```

Comment:
Why is it needed to cast m_casterUnit to Player?

Reply:
Good catch. I was just looking at the old conditionals and didn't think for myself.